contracts/CarbonVoteX.sol

```diff
@@ -0,0 +1,178 @@
+        require(pollExist(_pollId));
+
+        return polls[_pollId].gasSentByVoter[_voter];
+    }
+
+    // store the amount of gas sent by the voter 
+    function sendGas(bytes32 _pollId) public payable {
+        // poll must exits and not yet expired. 
+        require (pollExist(_pollId) && !pollExpired(_pollId));
+
+        emit _SendGas(msg.sender, _pollId);
+        polls[_pollId].gasSentByVoter[msg.sender] += msg.value;
+    }
+
+    // @param _pollId UUID (hash value) of a poll
+    // @param _startBlock starting block (unix timestamp) of the event
+    // @param _endBlock ending block (unix timestamp) of the event
+    // register a new poll
+    // Note that we do not allow re-registrations for the same UUID
+    function register(bytes32 _pollId, uint _startBlock, uint _endBlock, address _tokenAddr) public {
+        //Check if poll exists.
+        require (pollExist(_pollId) == false);
+        // check resonable endBlock and startBlock
+        require (_endBlock > block.number && _startBlock < _endBlock);
+        //Create a new poll and map the poll hashed value to the poll.
+        Poll memory poll = Poll({ startBlock: _startBlock,
+                                endBlock: _endBlock,
+                                pollId: _pollId,
+                                token: ERC20(_tokenAddr)});
+        polls[_pollId] = poll;
+        emit _Register(msg.sender, _pollId, _startBlock, _endBlock, _tokenAddr);
+    }
+
+    // @param pollId pollId (hash value) of a poll
+    // @param voter address of a voter    
+    // @param votes number of votes to write
+    // can only be called by the master address
+    function writeAvailableVotes(bytes32 _pollId, address _voter, uint _votes) public isMaster {
+        // poll must exits and not yet expired. 
+        require (pollExist(_pollId) && !pollExpired(_pollId));
+
+        polls[_pollId].availableVotes[_voter] += _votes; 
+        emit _WriteAvailableVotes(msg.sender, _pollId, _voter, _votes);
+    }
+
+    // @param pollId pollId (hash value) of a poll
+    // @param voter address of a voter
+    // returns the number of available votes
+    function readAvailableVotes(bytes32 _pollId, address _voter) view public returns (uint){
+        require (pollExist(_pollId));
+
+        return polls[_pollId].availableVotes[_voter];
+    }
+
+    // @param _pollId UUID (hash value) of a poll
+    // @param _voteMeta vote meta data
+    // @param _votes number of votes to redeem for voteMeta
+    // After the poll finishes, and have already called registerVotes()
+    // vote for "choice"
+    // deduct the total number of votes available by votes
+    function vote(bytes32 _pollId, bytes32 _choice, uint _votes) public{
+        // poll must exits and not yet expired. 
+        require (pollExist(_pollId) && !pollExpired(_pollId));
+        
+        // voter cannot vote more votes than it has.
+        assert (polls[_pollId].availableVotes[msg.sender] - _votes >= 0);
```

Comment:
change to require

Security risk: Note that polls[_pollId].availableVotes[msg.sender] - _votes can never < 0 since uint - uint = uint